src/main/java/lotto/domain/LottoTicket.java

```diff
@@ -0,0 +1,40 @@
+package lotto.domain;
+
+import java.util.List;
+
+public class LottoTicket {
+
+    public static final int LOTTO_SIZE = 6;
+
+    private final List<Integer> lottoNumbers;
+
+    public LottoTicket(NumberCreationStrategy strategy) {
+        this(strategy.createNumber());
+    }
+
+    public LottoTicket(List<Integer> lottoNumbers) {
+        validateLottoNumberLength(lottoNumbers);
+        this.lottoNumbers = lottoNumbers;
+    }
+
+    private void validateLottoNumberLength(List<Integer> lottoNumbers) {
+        if (lottoNumbers.size() != LOTTO_SIZE) {
+            throw new IllegalArgumentException("로또 번호는 6자리입니다.");
+        }
+    }
+
+    public int calculateSameNumberCount(LottoTicket lottoTicket) {
+        List<Integer> lottoNumber = lottoTicket.getLottoNumber();
+        return (int) this.lottoNumbers.stream()
+                .filter(lottoNumber::contains)
+                .count();
```

Comment:
getter로 값을 꺼내는 대신에, 메시지를 보내서 처리해보면 어떨까요?
```suggestion
        return (int) this.lottoNumbers.stream()
                .filter(number -> lottoTicket.hasNumber(number))
                .count();
```